### ISSUE TITLE
[PyTorch/Jax/common] Flex attention via cuDNN

### ISSUE THREAD

**Issue by cyanguwa** [NVIDIA/TransformerEngine#2492]
cuDNN flex attention offers more flexible support for attention masks or attention score modifications. We'd like to integrate this to TE on both the PyTorch side and Jax side.

References:
https://pytorch.org/blog/flexattention/
https://github.com/NVIDIA/cudnn-frontend/blob/0258951d4d512f4714eb1574496f4d57669b1b93/test/python/test_flexible_sdpa.py#L217
